contrib/gogearbox/gearbox/gearbox.go

```diff
@@ -0,0 +1,50 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2022 Datadog, Inc.
+
+// Package gearbox provides tracing functions for the gogearbox/gearbox package
+package gearbox
+
+import (
+	"context"
+
+	"gopkg.in/DataDog/dd-trace-go.v1/ddtrace"
+	"gopkg.in/DataDog/dd-trace-go.v1/ddtrace/ext"
+	"gopkg.in/DataDog/dd-trace-go.v1/ddtrace/tracer"
+
+	"github.com/gogearbox/gearbox"
+)
+
+const (
+	operationName = "http.request"
+)
+
+// Middleware acts as a gearbox middleware to provide tracing over incoming requests.
+func Middleware(ctx gearbox.Context) {
```

Comment:
We probably want to make this configurable. We very often add options for changing some of the behavior of integrations, especially web framework ones.

For example, please look at our gin middleware, which works similarly to this, but allows configuration:
https://github.com/DataDog/dd-trace-go/blob/main/contrib/gin-gonic/gin/option.go
https://github.com/DataDog/dd-trace-go/blob/main/contrib/gin-gonic/gin/gintrace.go#L22-L34

The only `Option` I  think we need to merge this is the analytics one.